File: javascript/alternateCapitals/alternateCapitals.js
// Given a string, return a copy of the original string that has every other character capitalized. (Capitalization should begin with the second character.)

// Input: “hello, how are your porcupines today?”
// Output: “hElLo, HoW ArE YoUr pOrCuPiNeS ToDaY?”

const alternateCapitals = function (string) {
  let index = 0
  let result = ""
  while (index < string.length) {

    if (index % 2 === 0) {
      result = result + string[index].toUpperCase()
    } else {
      result = result + string[index]
    }
    index += 1
  }
  return result
}

console.log(alternateCapitals("hello, how are your porcupines today?"))

File: ruby/algo_ladder/alternate_capitals/alternate_capitals.rb
# Given a string, return a copy of the original string that has every other character capitalized. (Capitalization should begin with the second character.)

# Input: “hello, how are your porcupines today?”
# Output: “hElLo, HoW ArE YoUr pOrCuPiNeS ToDaY?”

def alternate_capitals (input_string)
  result = ""
  index = 0
  while index < input_string.length

    if index % 2 == 1
      result << input_string[index].upcase
    else
      result << input_string[index]
    end
    index += 1
  end
  return result
end

p alternate_capitals("hello, how are your porcupines today?")